src/devtools/client/debugger/src/actions/sources/newSources.ts

```diff
@@ -63,28 +63,21 @@ function checkSelectedSource(cx: Context, sourceId: string): UIThunkAction {
     const state = getState();
     const pendingLocation = getPendingSelectedLocation(state);
 
-    if (!pendingLocation || !pendingLocation.url) {
+    if (!pendingLocation) {
       return;
     }
 
     const source = getSource(state, sourceId);
 
-    if (!source || !source.url) {
+    if (!source) {
       return;
     }
 
-    const pendingUrl = pendingLocation.url;
-    const rawPendingUrl = getRawSourceURL(pendingUrl);
-
-    if (rawPendingUrl === source.url) {
+    // Source IDs are not stable across Replays (or even across Replay sessions)
+    // so we compare via URL
+    if (pendingLocation.url === source.url) {
```

Comment:
This is the kind of place where `stableSourceId` would be nice. In truth, you can't know if a given file with the same URL actually contains this location. It could have totally different content, or it might be pretty-printed vs. not. If we had `kind` and `contentHash` as well, we would have true guarantees.